### PR TITLE
fix(demultiplexing )Remove demux dir before starting a new demux job

### PR DESCRIPTION
**Pull request by Vince-janv**
## Description
Before running demultiplexing the output directory needs to be removed in preparation for the new data.

This PR removes two pre-checks before demultiplexing and, if the command is not run with dry-run, also removes the output directory before submitting the new demultiplexing job.

With the new behaviour we will not check anything in the output directory before starting demultiplexing. We will rely on the the demuxstarted.txt file to decide wether to start demultiplexing (and removing the output dir).

### Added
- Method `prepare_output_directory` to both remove and create output directory

### Changed

- Output dir creation moved from `start_demultiplexing` to new function `prepare_output_directory`
- Method `create_demultiplexing_output_dir` now only takes one argument

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b delete-demux-dir
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Review by henrikstranneheim — APPROVED** [review]
👍

**Review by diitaz93 — APPROVED** [review]
👍

File: cg/cli/demultiplex/demux.py [hunk 1]
```diff
@@ -59,6 +59,7 @@ def demultiplex_all(context: CGConfig, flow_cells_directory: click.Path, dry_run
             continue
 
         if not dry_run:
+            demultiplex_api.remove_demultiplexing_output_directory(flow_cell)
```
Comment (ChrOertlin) [inline cg/cli/demultiplex/demux.py:62]: can we put this into the `demultiplex_api.start_demultiplexing(flow_cell=flow_cell)` function just before where we make the directory? 
Might even be worth considering to extract making and deleting the directory a separate function together. 
That function can check if exist, delete and create new.

Comment (Vince-janv) [reply cg/cli/demultiplex/demux.py:67]: I'm would like to keep the deleting out of `start_demultiplexing` as it rather prepares before the start. But do you suggest extracting the directory creation out from `start_demultiplexing`?

Comment (ChrOertlin) [reply cg/cli/demultiplex/demux.py:67]: My suggestion would: 1. make the CLI cleaner. 2. Keep logic regarding the demux dir together. The idea was to move the deletion/creation/existence check closer together instread of having one in the CLI and the other calls in the start demultiplexing. Can be outside or inside the start demux call.

Comment (Vince-janv) [reply cg/cli/demultiplex/demux.py:67]: I created the method `prepare_output_directory` which groups together delete and create methods. Put it outside of the start command. Could you take a quick look?

Comment (ChrOertlin) [reply cg/cli/demultiplex/demux.py]: Looks good to me"

**Comment by Vince-janv** [comment]
Tests pass in stage

**Comment by Vince-janv** [comment]
Deployed in production

**Pull request merged**